packages/@uppy/webcam/src/Webcam.test.ts

```diff
@@ -1,43 +1,55 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
 import { describe, expect, it } from 'vitest'
 import Uppy from '@uppy/core'
-import Webcam from '../lib/index.js'
+import Webcam from './index.ts'
 
 describe('Webcam', () => {
   describe('_getMediaRecorderOptions', () => {
     it('should not have a mimeType set if no preferences given', () => {
+      // @ts-ignore
       globalThis.MediaRecorder = {
         isTypeSupported: () => true,
       }
 
-      const uppy = new Uppy().use(Webcam)
+      const uppy = new Uppy<any, any>().use(Webcam)
       expect(
-        uppy.getPlugin('Webcam').getMediaRecorderOptions().mimeType,
+        (uppy.getPlugin('Webcam') as Webcam<any, any>).getMediaRecorderOptions()
+          .mimeType,
       ).not.toBeDefined()
     })
 
     it('should use preferredVideoMimeType', () => {
+      // @ts-ignore
       globalThis.MediaRecorder = {
         isTypeSupported: (ty) => ty === 'video/webm',
       }
 
-      const uppy = new Uppy().use(Webcam, { preferredVideoMimeType: 'video/webm' })
+      const uppy = new Uppy().use(Webcam, {
+        preferredVideoMimeType: 'video/webm',
+      } as ConstructorParameters<typeof Webcam>[1])
```

Comment:
@Murderlon @nickrttn  we probably want to find a way to avoid the need for this `as ConstructorParameters<typeof Webcam>[1]` that is currently required.

For reference, the error if I remove it is the following:

```
packages/@uppy/webcam/src/Webcam.test.ts:89:14 - error TS2345: Argument of type 'typeof Webcam' is not assignable to parameter of type 'new (uppy: Uppy<Meta, Body>, opts?: { preferredVideoMimeType: string; } | undefined) => UIPlugin<{ preferredVideoMimeType: string; }, Meta, Body, Record<string, unknown>> | BasePlugin<...>'.
  Types of construct signatures are incompatible.
    Type 'new <M extends Meta, B extends Body>(uppy: Uppy<M, B>, opts?: Partial<WebcamOptions<M, B>> | undefined) => Webcam<M, B>' is not assignable to type 'new (uppy: Uppy<Meta, Body>, opts?: { preferredVideoMimeType: string; } | undefined) => UIPlugin<{ preferredVideoMimeType: string; }, Meta, Body, Record<string, unknown>> | BasePlugin<...>'.
      Type 'Webcam<Meta, Body>' is not assignable to type 'UIPlugin<{ preferredVideoMimeType: string; }, Meta, Body, Record<string, unknown>> | BasePlugin<{ preferredVideoMimeType: string; }, Meta, Body, Record<...>>'.
        Type 'Webcam<Meta, Body>' is not assignable to type 'UIPlugin<{ preferredVideoMimeType: string; }, Meta, Body, Record<string, unknown>>'.
          The types of 'opts.preferredVideoMimeType' are incompatible between these types.
            Type 'string | null' is not assignable to type 'string'.
              Type 'null' is not assignable to type 'string'.

89       }).use(Webcam, {
                ~~~~~~
```

The method is defined here: https://github.com/transloadit/uppy/blob/218c6d160b0e4b4db0fefb62f968d959a5f95121/packages/@uppy/core/src/Uppy.ts#L1666-L1669

Reply:
<img width="868" alt="Screenshot 2024-01-18 at 17 38 55" src="https://github.com/transloadit/uppy/assets/2504906/408f2a93-43a1-4082-bae7-ea04d018963c">

works.

I would argue you need to export the options type for each plugin as it is part of the public API. Any type the user interfaces with needs to be exportable as well for things like this.

Reply:
> In general, this probably means the inferral of the generics used in `uppy.use()` could be improved to infer this so passing the generic parameters wouldn't be necessary in most cases

Yes exactly, because I don't see how passing two generics is better than the single `as ConstructorParameters<typeof Webcam>[1])`, in both cases it's not great for readability (and annoying to type).